engine/server/sv_client.c

```diff
@@ -1363,6 +1351,18 @@ void SV_New_f( sv_client_t *cl )
 	// refresh userinfo on spawn
 	SV_RefreshUserinfo();
 
+	// get the game a chance to reject this connection or modify the userinfo
+	if( !SV_ClientConnect( cl->edict, cl->userinfo ) )
+	{
+		if( *Info_ValueForKey( cl->userinfo, "rejmsg" ))
+			Netchan_OutOfBandPrint( NS_SERVER, cl->netchan.remote_address, "errormsg\n%s\nConnection refused.\n", Info_ValueForKey( cl->userinfo, "rejmsg" ));
+		else Netchan_OutOfBandPrint( NS_SERVER, cl->netchan.remote_address, "errormsg\nConnection refused.\n");
```

Comment:
@mittorn sending errormsg to 0.18 client would not break anything. 
At least, they will see "Bad connectionless packet: errormsg".